File: app/src/main/java/org/chupik/redditringtest/Post.java
package org.chupik.redditringtest;

import androidx.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

public class Post {

    String name;
    String title;
    String author;
    @SerializedName("created_utc")
    long date;
    String thumbnail;
    @Nullable
    @SerializedName("url")
    String fullImage;
    @SerializedName("num_comments")
    long commentsNumber;
    String permalink;
    Preview preview;

    public Post(String name, String title, String author, long date, String thumbnail, @Nullable String fullImage, long commentsNumber, String permalink) {
        this.name = name;
        this.title = title;
        this.author = author;
        this.date = date;
        this.thumbnail = thumbnail;
        this.fullImage = fullImage;
        this.commentsNumber = commentsNumber;
        this.permalink = permalink;
    }

    public long getDateInMiliseconds() {
        return date * 1000;
    }

    public String getFullPermaLink() {
        return "https://www.reddit.com" + permalink;
    }

    @Nullable
    public String getFullImageIfEnabled() {
        if (preview != null && preview.getEnabled())
            return fullImage;
        else
            return null;
    }
}

File: app/src/main/java/org/chupik/redditringtest/Post.kt
package org.chupik.redditringtest

import com.google.gson.annotations.SerializedName

class Post(
        val name: String,
        val title: String,
        val author: String,
        @SerializedName("created_utc")
        val date: Long,
        val thumbnail: String?,
        @SerializedName("url")
        val fullImage: String?,
        @SerializedName("num_comments")
        val commentsNumber: Long,
        val permalink: String) {

    var preview: Preview? = null

    val dateInMiliseconds: Long
        get() = date * 1000

    val fullPermaLink: String
        get() = "https://www.reddit.com$permalink"

    val fullImageIfEnabled: String?
        get() = if (preview?.enabled == true) fullImage else null

}